yellowbrick/classifier/boundaries.py

```diff
@@ -0,0 +1,457 @@
+        """
+        The fit method is the primary drawing input for the decision boundaries
+        visualization since it has both the X and y data required for the
+        viz and the transform method does not.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the visualizer
+        """
+        X = self._select_feature_columns(X)
+
+        # Assign each class a unique number for drawing
+        if self.classes_ is None:
+            self.classes_ = {
+                label: str(kls_num)
+                for kls_num, label in enumerate(np.unique(y))
+            }
+            self.class_labels = None
+        elif len(set(y)) == len(self.classes_):
+            self.classes_ = {
+                label: str(kls_num)
+                for kls_num, label in enumerate(self.classes_)
+            }
+            self.class_labels = dict(zip(set(y), self.classes_))
+        else:
+            raise YellowbrickTypeError(
+                """Number of classes must be the same length of number of
+                target y"""
+            )
+
+        # ensure that only
+        self.estimator.fit(X, y)
+
+        # Plot the decision boundary. For that, we will assign a color to each
+        # point in the mesh [x_min, x_max]x[y_min, y_max].
+        x_min, x_max = X[:, 0].min() - 1, X[:, 0].max() + 1
+        y_min, y_max = X[:, 1].min() - 1, X[:, 1].max() + 1
+
+        # Create the axes if they don't exist
+        if self.ax is None:
+            self.ax = plt.gca(xlim=[x_min, x_max], ylim=[y_min, y_max])
+
+        # set the step increment for drawing the boundary graph
+        x_step = (x_max - x_min) * self.step_size
+        y_step = (y_max - y_min) * self.step_size
+
+        self.xx, self.yy = np.meshgrid(
+            np.arange(x_min, x_max, x_step), np.arange(y_min, y_max, y_step))
+
+        # raise Exception(self.yy.ravel().shape)
+        Z = self.estimator.predict(np.c_[self.xx.ravel(), self.yy.ravel()])
+        self.Z_shape = Z.reshape(self.xx.shape)
+        return self
+
+    def draw(self, X, y=None, **kwargs):
+        """
+        Called from the fit method, this method creates a decision boundary
+        plot, and if self.scatter is True, it will scatter plot that draws
+        each instance as a class or target colored point, whose location
+        is determined by the feature data set.
+        """
+        # ensure that if someone is passing in another X such as X_test, that
+        # features will be properly handled
+        X = self._select_feature_columns(X)
+
+        num_colors = len(self.classes_) * 2
```

Comment:
Is this one color for the scatter point and one for the background for each class?